lib/pure/json.nim

```diff
@@ -617,7 +617,7 @@ proc newJObject*(): JsonNode =
   ## Creates a new `JObject JsonNode`
   new(result)
   result.kind = JObject
-  result.fields = @[]
+  result.fields = initTable[string, JsonNode]()
```

Comment:
using `initTable[string, JsonNode](4)` here and in 2 more places will save memory and improve performance for small objects